### PR TITLE
[WIP] Rust interface

### PR DESCRIPTION
**Pull request by tholop**
Let's separate the core Rust functionality from the Python wrapper!

Some things to do in this PR:
- Refactoring some traits (FSSKey doesn't need to be Numpy-compatible)
- Exposing cleaner interfaces
- Generic DPF and DIF functionalities instead of AriaNN-tailored code (e.g. in AriaNN, we use a special case of DPF where α is random and additively shared, while β = 1).
- First step to make Sycret usable as a standalone Rust crate :crab:

### CODE REVIEW

File: src/le.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use std::fmt;
 use std::num::Wrapping;
```
Comment (arturomf94) [inline src/le.rs:4]: We've got lots of warnings for unused imports! 😱

Comment (tholop) [reply src/le.rs]: I just discovered `cargo fix` :star_struck:

**Comment by arturomf94** [comment]
Was `ae94fa7 ` all cargo fix!?

**Comment by tholop** [comment]
> Was `ae94fa7 ` all cargo fix!?

Yep, 100%

**Pull request merged**